### PR TITLE
Add Trader Joe v2.1 on Avalanche to dex.trades and dex.trades_beta

### PR DESCRIPTION
**Pull request by chef-seaweed**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by Hosuke** [comment]
Hi @chef-seaweed, you want to use the macro approach to implement `trader_joe_v2_1_avalanche_c.base_trades` as well?

You can find some sample PRs in https://github.com/duneanalytics/spellbook/issues/4759

**Comment by chef-seaweed** [comment]
> Hi @chef-seaweed, you want to use the macro approach to implement `trader_joe_v2_1_avalanche_c.base_trades` as well?
> 
> You can find some sample PRs in #4759

Hi @Hosuke, I can try to implement it but curious do I open a new PR for this? thanks

**Comment by chef-seaweed** [comment]
Hi @Hosuke seems like this is ready to be reviewed, thanks 🙏

File: macros/models/_sector/dex/trader_joe_compatible_trades.sql
Comment (Hosuke) [inline macros/models/_sector/dex/trader_joe_compatible_trades.sql:1]: Nice macro.

**Comment by Hosuke** [comment]
I just added other versions of trader_joe base trades into this PR.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @chef-seaweed

File: models/_sector/dex/trades/arbitrum/platforms/trader_joe_v2_arbitrum_base_trades.sql
Comment (jeff-dude) [inline models/_sector/dex/trades/arbitrum/platforms/trader_joe_v2_arbitrum_base_trades.sql:1]: looks like `v2` is another good candidate for a macro. it appears arbitrum/avalanche/bnb all use the same logic. similar to how we did `uniswap_compatible_trades` macro, we can combine all trader joe macros into one file and just have a different macro name for `v2` within

can we update to reflect that design?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

**Comment by chef-seaweed** [comment]
Screenshot 2023-12-06 at 12 12 43 PM
Not sure why the error, @Hosuke @jeff-dude can you help take a look? thank you 🙏

**Comment by chef-seaweed** [comment]
Screenshot 2023-12-06 at 12 12 43 PM
Hi @Hosuke @jeff-dude, could you help take a look why the errors? I'm a bit clueless, thank you 🙏

File: macros/models/_sector/dex/trader_joe_compatible_trades.sql [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    , Factory_evt_PoolCreated = null
+    , pair_column_name = 'LBPair'
+    )
+%}
+WITH dexs AS
+(
+    SELECT
+        t.evt_block_number AS block_number
+        , t.evt_block_time AS block_time
+        , t.to AS taker
```
Comment (Hosuke) [inline macros/models/_sector/dex/trader_joe_compatible_trades.sql:15]:
```suggestion
        , t.recipient AS taker
```

**Comment by Hosuke** [comment]
> Screenshot 2023-12-06 at 12 12 43 PM Hi @Hosuke @jeff-dude, could you help take a look why the errors? I'm a bit clueless, thank you 🙏

Solved.

File: macros/models/_sector/dex/trader_joe_compatible_trades.sql (hunk 1 above)
Comment (chef-seaweed) [reply macros/models/_sector/dex/trader_joe_compatible_trades.sql]: Ah right, totally missed that 😅
Thank you for the help 🙏

**Review by jeff-dude — APPROVED** [review]
nice work here 🔥

**Pull request merged**